Explore relationship between A and B

ajlee21 opened the pull request:
This PR adds a new notebook "explore_relationship_AandB.ipynb" that continues to explore the data and what our model is learning.

Previously we found that our model is in fact capturing genes from group A and genes from group B

Now, we want to know if our model is capturing the relationship between the two groups.=

danich1 commented on scripts/nbconverted/explore_relationship_AandB.py:
```diff
@@ -0,0 +1,585 @@
+# Get the means of B genes
+
+# Convert dataframe with gene ids to list
+geneSetB_ls = [l[0] for l in geneSetB.values.tolist()]
```
Couldnt u just do geneSetB.id.values.tolist()? would save u time in extracting the ids from dataframe.

ajlee21 replied:
Thanks for catching this.  I had been fiddling around trying to use pandas data structures because I didn't think it should be so complex.  When I tried ```geneSetB.values.tolist()``` this returned to me a list of lists so thus the list comprehension.

danich1 commented on scripts/nbconverted/explore_relationship_AandB.py:
```diff
@@ -0,0 +1,585 @@
+
+
+# Plot
+sns.regplot(x='gene A untransformed',
```
Whats going on with this plot? Is it suppose to show that some gene are upregulated given gene A has high expression? Why the mean of gene B an not just regular values?

ajlee21 replied:
I will add a comment.  This plot is meant to show the original simulated dataset that is input into the model.  It is meant to serve as a reference to compare what the data looks like before the data goes into the model and is transformed.

danich1 commented on scripts/nbconverted/explore_relationship_AandB.py:
```diff
@@ -0,0 +1,585 @@
+                                       percent_high,
+                                       out_dir):
+    """
+    Description here
```
Can we fill this doc string in so one can know what is going on with the variables in this function?

ajlee21 replied:
Forgot to add the description.  I will add this

danich1 commented on scripts/nbconverted/explore_relationship_AandB.py:
```diff
@@ -0,0 +1,585 @@
+# Get the means of B genes
+
+# Convert dataframe with gene ids to list
+geneSetB_ls = [l[0] for l in geneSetB.values.tolist()]
```
refer to genesetb dataframe comment.

ajlee21 replied:
Fixed as noted above

danich1 commented on scripts/nbconverted/explore_relationship_AandB.py:
```diff
@@ -0,0 +1,585 @@
+
+
+# Plot
+sns.regplot(x='gene A untransformed',
```
Yay it is linear! is that expected? I.e. is this a good result?

ajlee21 replied:
This plot shows that the relationship that is learned by the VAE appears to be linear, which is NOT the relationship that we put into the dataset (remember we put a step function relationship).  So now the question is why the VAE is learning a linear relationship between A and B genes?  Or is this an artifact of how we plotted the data?

danich1 commented on scripts/nbconverted/explore_relationship_AandB.py:
```diff
@@ -0,0 +1,585 @@
+lut = dict(zip(threshold_groups.unique(), "rbgm"))
+row_colors = threshold_groups.map(lut)
+
+sns.clustermap(data_encoded,
```
Can we change the cmap to viridis or something with a bit more cooler tone? On the other hand it looks like there isn't a change between samples that were assigned different groups. Is this a problem? Is this what you expect?

danich1 commented on scripts/nbconverted/explore_relationship_AandB.py:
```diff
@@ -0,0 +1,585 @@
+trend_feature_df = pd.merge(original_A_exp.to_frame('gene A untransformed'),
+                      feature_values.to_frame('values of feature {}'.format(top_feature)),
+                      left_index=True, right_index=True)
+trend_feature_df.head()
```
How do you go from your top feature being 19 to 106? Is it because 19 is too negative?

ajlee21 replied:
There are 5 top features that showed the largest ABSOLUTE difference (regardless of sign).  The top feature is 19.  106 is the 3rd most different feature.  I was manually playing around with the values by changing the index in ```top_feature = top_features.index[2]```

danich1 commented on scripts/nbconverted/explore_relationship_AandB.py:
```diff
@@ -0,0 +1,585 @@
+
+
+# Get genes associated with top feature
+top_feature = int(top_features.index[0])
```
Is this 106 or 19?

ajlee21 replied:
See above comment.  Thanks for catching the inconsistency.  There is a lot of manual playing around here that I did to explore the dataset so its a little clunky

ajlee21 commented:
I think I have address all your comments.  Please let me know if you would like to sit down to walk through the analysis and especially the interpretations.  All feedback is welcomed and helpful!

danich1 commented:
Just forgot to change the heatmap's color scheme. Everything else looks good. Will be in on Monday to quickly finalize in person.

ajlee21 commented:
Based on discussion with @danich1 I have made the following changes:
1.  Changed color scheme of the heatmap so its easier to look at
2.  Added additional comments to help clarify the exploratory analysis being done
3.  Made a new plot of transformed A vs transformed B in addition to untransformed A vs transformed B (they both look linear)

Please let me know if there are additional comments or suggestions
